### PR TITLE
fix PercolatorAdapter

### LINKED ISSUE #3188: Percolator reports all peptides with 1 score in OpenMS 2.3 with labeled samples
I have been trying to integrate Percolator to my workflow in OpenMS 2.3 (TOPPAS on Windows) but so far have not succeed. This issue might be related to https://github.com/OpenMS/OpenMS/issues/3071.

My workflow is simple and is just an adaptation from a post of Mathew The on the Gitter OpenMS/Knime chat, but has not been succesful.

MSGFPlusAdapter —> PeptideIndexer —> PercolatorAdapter

I can reproduce Mathew The results, with the yeast test file, however I have no luck with my test file which is a Formaldehyde labeled sample (K and n-term labeled) measured in a Qexactive. The changes to the workflow thus are only adding the fixed modification (CAM, K and n-term dimethylation).

I have also tried to do the search with a database without decoys and let the MSGFPlusAdapter add the decoys (and of course removing the peptide indexer node). I also have done it with the XtandemAdapter, with similar results. 

Percolator runs through, but using MSGF in the training portion reports 0 PSMs at q < 0.01, even though before training recognizes both targets and decoys, and finally reports "0 suitable peptides found". If I use the mzid output of msgfplus as input instead, it does recognize PSMs below q 0.01 at the training but it also ends with 0 suitable peptides. Looking at the produced idxml file, the first part of the file, protein section all seem to have a score of 0, whereas at the peptide level all get a score of 1. Running it with xTandem, does reports PSMs below q 0.01 at training but ends with the same result (All peptides with score 1). 

If I take the MSGFplusAdapter mzid output and do msgf2pin and percolator on the command line in the same PC, percolator runs fine, reports both peptides at training (at all iterations) and finds suitable peptides, with probabilities from 0-1 in the end.

Setting debugging at 5 in PercolatorAdapter to keep the intermediate files I get the pin input file (which I guess comes from the idxml) and what I guess the target and decoy output. Running percolator on the command line with that pin file (Of both xTandem and MSGFplus) also runs through, although it does not find peptides in all iterations at training with msgf input thus resets the vector, however in the end finds peptides below q 0.01.

Comparing both files (msgf2pin/percolator and pin form debugging) shows that msgf2pin or tandem2pin annotates the labeling as e.g. K.[UNIMOD:36] whereas the openms adaptor annotates them as K..(Dimethyl) (2 dots after K), and the one form msgf2pin does not annotate the CAM. The other differences is a lack of the second line in the one produced by the PercolatorAdapter (which includes the DefaultDirection). 
After parsing the file to the same notation (UNIMOD) there is a 81% overlap on the peptides at q<0.01 when comapring both files. 

Any thoughts?

I have attached a folder with the workflow and some outputs.

[msgfperc_git.zip](https://github.com/OpenMS/OpenMS/files/1652136/msgfperc_git.zip)

### PR DESCRIPTION
**Pull request by lars20070**
Set dots in the AA sequence only, if no N/C terminal mods are present.
Fixes #3188.

### CODE REVIEW

**Comment by timosachsenberg** [comment]
look good

**Comment by timosachsenberg** [comment]
maybe add some comment to it

**Comment by lars20070** [comment]
@algom `Percolator` now runs through on our dimethyl data. Let's test how much difference it makes after I am back.

**Comment by lars20070** [comment]
rebuild jenkins

File: src/topp/MSGFPlusAdapter.cpp [hunk 1]
```diff
@@ -181,7 +181,7 @@ class MSGFPlusAdapter :
     registerIntOption_("matches_per_spec", "<num>", 1, "Number of matches per spectrum to be reported (MS-GF+ parameter '-n')", false);
     setMinInt_("matches_per_spec", 1);
 
-    registerFlag_("add_features", "Output additional features - needed e.g. by Percolator (default: basic scores only; MS-GF+ parameter '-addFeatures')", false);
+    registerFlag_("add_features", "Output additional features - needed e.g. by Percolator (default: basic scores only; MS-GF+ parameter '-addFeatures')", true);
```
Comment (timosachsenberg) [inline src/topp/MSGFPlusAdapter.cpp]: doesn't work that way. needs to be a string with true false

File: src/topp/MSGFPlusAdapter.cpp [hunk 2]
```diff
@@ -181,8 +181,9 @@ class MSGFPlusAdapter :
     registerIntOption_("matches_per_spec", "<num>", 1, "Number of matches per spectrum to be reported (MS-GF+ parameter '-n')", false);
     setMinInt_("matches_per_spec", 1);
 
-    registerFlag_("add_features", "Output additional features - needed e.g. by Percolator (default: basic scores only; MS-GF+ parameter '-addFeatures')", true);
-
+    registerStringOption_("add_features", "<bool>", "true", "Output additional features - needed e.g. by Percolator (default: basic scores only; MS-GF+ parameter '-addFeatures')");
```
Comment (lars20070) [inline src/topp/MSGFPlusAdapter.cpp]: @timosachsenberg That builds, but is not correct. Could you please point out an example?

Comment (timosachsenberg) [reply src/topp/MSGFPlusAdapter.cpp]: something like in this snippet:
```
registerStringOption_("bla", "<true/false>", "true", "bla", false, false);
setValidStrings_("bla", ListUtils::create<String>("true,false"));
```

Comment (timosachsenberg) [reply src/topp/MSGFPlusAdapter.cpp]: also update description to not talk about flag etc.

Comment (lars20070) [reply src/topp/MSGFPlusAdapter.cpp]: Why? - Can I kick out the exception?

**Comment by hroest** [comment]
:+1: for merge

**Pull request merged**